ext/standard/dl.c

```diff
@@ -205,6 +205,10 @@ PHPAPI int php_load_extension(const char *filename, int type, int start_now)
 		return FAILURE;
 	}
 	module_entry = get_module();
+	if (zend_hash_str_exists(&module_registry, module_entry->name, strlen(module_entry->name))) {
+		zend_error(E_CORE_WARNING, "Module \"%s\" is already loaded", module_entry->name);
+		return FAILURE;
```

Comment:
We may need to `DL_UNLOAD(handle)` here. I believe that `handle` is refcounted, so it will not actually unload the module here, but will allow the module to be unloaded later.

Reply:
Ah, right. Thank you!